tensorboardX/writer.py

```diff
@@ -266,11 +266,10 @@ def __init__(self, log_dir=None, comment='', **kwargs):
             current_time = datetime.now().strftime('%b%d_%H-%M-%S')
             log_dir = os.path.join(
                 'runs', current_time + '_' + socket.gethostname() + comment)
+        self.log_dir = log_dir
 
         if 'purge_step' in kwargs.keys():
             most_recent_step = kwargs.pop('purge_step')
-            if not os.path.exists(log_dir):
```

Comment:
Because `os.path.exists` basically requires that the log_dir is a local file, but in the S3 case that's not true. The alternative is to have it route through some filesystem abstraction (we are really moving in the direction of a `tf.gfile` replacement that will also be used in the non-TensorFlow version of TensorBoard), but it seemed like the line was just informationally.